Dockerfile

```diff
@@ -5,15 +5,10 @@ RUN apt-get update && apt-get install -y make git unzip
 
 # should stay consistent with the version we use in Circle builds
 ARG PROTOC_VERSION=3.5.0
-# make a directory to do these operations in
-RUN export PROTOC_TMP_DIR=protoc && mkdir -p $PROTOC_TMP_DIR && cd $PROTOC_TMP_DIR \
-  # download the pre-built protoc binary
-  && curl --silent --show-error --location --output protoc.zip \
-    https://github.com/google/protobuf/releases/download/v$PROTOC_VERSION/protoc-$PROTOC_VERSION-linux-x86_64.zip \
-  # move the binary to /bin. move the well-known types ot /usr/local/include
-  && unzip protoc.zip && mv bin/protoc /bin/protoc && mv include/* /usr/local/include \
-  # remove all of the installation files
-  && cd .. && rm -rf $PROTOC_TMP_DIR
+RUN curl --silent --show-error --location --output protoc.zip \
+  https://github.com/google/protobuf/releases/download/v$PROTOC_VERSION/protoc-$PROTOC_VERSION-linux-x86_64.zip \
+  && unzip protoc.zip -x readme.txt -d /usr/local \
```

Comment:
I kinda preferred the explicit `mv`s above, since it at least ensured we didn't copy whatever was in the archive, but only some files we cared about. IMO that outweighs whatever I/O we waste with the current version?

Reply:
There is nothing else in there (except for the readme.txt which we skip):
```
Archive:  protoc.zip
   creating: /usr/local/include/google/
   creating: /usr/local/include/google/protobuf/
  inflating: /usr/local/include/google/protobuf/struct.proto  
  inflating: /usr/local/include/google/protobuf/type.proto  
  inflating: /usr/local/include/google/protobuf/descriptor.proto  
  inflating: /usr/local/include/google/protobuf/api.proto  
  inflating: /usr/local/include/google/protobuf/empty.proto  
   creating: /usr/local/include/google/protobuf/compiler/
  inflating: /usr/local/include/google/protobuf/compiler/plugin.proto  
  inflating: /usr/local/include/google/protobuf/any.proto  
  inflating: /usr/local/include/google/protobuf/field_mask.proto  
  inflating: /usr/local/include/google/protobuf/wrappers.proto  
  inflating: /usr/local/include/google/protobuf/timestamp.proto  
  inflating: /usr/local/include/google/protobuf/duration.proto  
  inflating: /usr/local/include/google/protobuf/source_context.proto  
  inflating: /usr/local/bin/protoc  
```

I have also rechecked it with a latest version, 3.6.1:
```
Archive:  protoc.zip
   creating: /usr/local/include/google/
   creating: /usr/local/include/google/protobuf/
  inflating: /usr/local/include/google/protobuf/wrappers.proto  
  inflating: /usr/local/include/google/protobuf/field_mask.proto  
  inflating: /usr/local/include/google/protobuf/api.proto  
  inflating: /usr/local/include/google/protobuf/struct.proto  
  inflating: /usr/local/include/google/protobuf/descriptor.proto  
  inflating: /usr/local/include/google/protobuf/timestamp.proto  
   creating: /usr/local/include/google/protobuf/compiler/
  inflating: /usr/local/include/google/protobuf/compiler/plugin.proto  
  inflating: /usr/local/include/google/protobuf/empty.proto  
  inflating: /usr/local/include/google/protobuf/any.proto  
  inflating: /usr/local/include/google/protobuf/source_context.proto  
  inflating: /usr/local/include/google/protobuf/type.proto  
  inflating: /usr/local/include/google/protobuf/duration.proto  
  inflating: /usr/local/bin/protoc   
```

I suppose the creators of this .zip file are taking care of what's in there :laughing:

Reply:
Doesn't mean there'll never be anything else :) Sure, it's unlikely, but then for such a small benefit, why bother?

That being said, I don't feel super strongly about this, so as long as @dperny is okay with it, why not!

Reply:
@kolyshkin actually looking at `unzip`'s man page:
```
SYNOPSIS
       unzip [-Z] [-cflptTuvz[abjnoqsCDKLMUVWX$/:^]] file[.zip] [file(s) ...]  [-x xfile(s) ...] [-d exdir]
[...]
       [file(s)]
              An optional list of archive members to be processed, separated by spaces.  (VMS versions compiled with  VMSCLI
              defined  must  delimit  files with commas instead.  See -v in OPTIONS below.)  Regular expressions (wildcards)
              may be used to match multiple members; see above.  Again, be sure to quote expressions that would otherwise be
              expanded or modified by the operating system.
```
so let's white-list the bin and the includes instead of blacklisting the readme?